src/main/java/com/redhat/cloud/policies/app/rest/PolicyCrudService.java

```diff
@@ -607,33 +651,47 @@ public Response setEnabledStateForPolicies(@QueryParam("enabled") boolean should
         registry.counter("policies_ui_setEnabledStateForPolicies", new Tag("account", user.getAccount())).inc();
 
         List<UUID> changed = new ArrayList<>(uuids.size());
-        try {
+        if (lightweightEngineConfig.isEnabled()) {
             for (UUID uuid : uuids) {
                 Policy storedPolicy = Policy.findById(user.getAccount(), uuid);
-                boolean wasChanged = false;
                 if (storedPolicy != null) {
-                    try {
-                        if (shouldBeEnabled) {
-                            engine.enableTrigger(storedPolicy.id, user.getAccount());
-                        } else {
-                            engine.disableTrigger(storedPolicy.id, user.getAccount());
+                    storedPolicy.isEnabled = shouldBeEnabled;
+                    storedPolicy.setMtimeToNow();
+                    storedPolicy.persist();
+                    changed.add(uuid);
+                }
+            }
+            accountLatestUpdateRepository.setLatestToNow(user.getAccount());
```

Comment:
I guess this could count as an optimization, not something i would expect to happen often so feel free to ignore it

```suggestion
                    if (storedPolicy.isEnabled != shouldBeEnabled) {
                        storedPolicy.isEnabled = shouldBeEnabled;
                        storedPolicy.setMtimeToNow();
                        storedPolicy.persist();
                        changed.add(uuid);
                    }
                }
            }
            if (!changed.isEmpty()) {
                accountLatestUpdateRepository.setLatestToNow(user.getAccount());
            }
```